### ISSUE TITLE
Markdown reference in section heading of changelog not parsed correctly

### ISSUE THREAD

**Issue by hedtke** [doxygen/doxygen#9449]
**Describe the bug**
We use the format https://keepachangelog.com/en/1.0.0/ to manage our changelog in a markdown file.
This format uses references to link to specific versions in the VCS.
We add this file via `INPUT=` to the doxygen sources.
The resulting HTML shows the raw HTML syntax, not the headline with links

**Expected behavior**
headlines with links

**Screenshots**
![image](https://user-images.githubusercontent.com/726611/178999030-cd0fa51a-4b6f-43d8-ad6f-1212ee557134.png)

**To Reproduce**
[doxygenbug.zip](https://github.com/doxygen/doxygen/files/9112399/doxygenbug.zip)

**Version**
1.9.4

**Comment by albert-github** [comment]
With doxygen 1.9.4 I see the headings as indicated.
With the current doxygen version (1.15.0) the headers are correctly formatted.

I think the problem has been solved in the mean time

**Comment by hedtke** [comment]
Indeed, the problem is solved. Already in 1.14, maybe even earlier.
